workflow/session.go

```diff
@@ -47,9 +47,20 @@ type (
 	SessionOptions = internal.SessionOptions
 )
 
-// ErrSessionFailed is the error returned when user tries to execute an activity but the
-// session it belongs to has already failed
-var ErrSessionFailed = internal.ErrSessionFailed
+var (
+	// ErrSessionFailed is the error returned when user tries to execute an activity but the
+	// session it belongs to has already failed
+	ErrSessionFailed = internal.ErrSessionFailed
+
+	// SessionStateOpen means the session worker is heartbeating and new activities will be schedule on the session host.
+	SessionStateOpen = internal.SessionStateOpen
+
+	// SessionStateClosed means the session was closed by the workflow and new activities will not be scheduled on the session host.
+	SessionStateClosed = internal.SessionStateClosed
+
+	// SessionStateFailed means the session worker was detected to be down and the session cannot be used to schedule new activities.
+	SessionStateFailed = internal.SessionStateFailed
```

Comment:
How about the `SessionState` type itself?

Reply:
hmmm yes I guess I should. Originally I didn't think I needed to, but If someone wants to make a function that takes the state as a parameter it would be useful to expose.

Reply:
I just want to make sure this session state var is updated on worker crash before activity failure is returned. If not and the session appears open but activity fails, we may need to document that.